src/Widgets/NotificationEntry.vala

```diff
@@ -231,6 +265,28 @@ public class Notifications.NotificationEntry : Gtk.ListBoxRow {
 
         show_all ();
 
+        button_release_event.connect (() => {
+            unowned var action_group = get_action_group (ACTION_GROUP_PREFIX);
+            if (action_group.has_action (notification.default_action)) {
+                notification.app_info.launch_action (Notification.DEFAULT_ACTION, new GLib.AppLaunchContext ());
+                action_group.activate_action (notification.default_action, null);
+                clear (false);
+            } else if (notification.actions.length == 0) {
+                try {
+                    notification.app_info.launch (null, null);
+                } catch (Error e) {
+                    critical ("Unable to launch app: %s", e.message);
+                }
+                clear ();
+            } else {
+                return Gdk.EVENT_STOP;
+            }
+
+            activate ();
+
+            return Gdk.EVENT_STOP;
```

Comment:
this cannot be reached.

Reply:
Hmm I think it can be reached as the two previous `if/else if` don't return but just continue or did I miss something?

Reply:
Oh right, i missed that both first if/else if don't return.